types/module/module.go

```diff
@@ -222,6 +222,11 @@ type HasConsensusVersion interface {
 	ConsensusVersion() uint64
 }
 
+type PreBeginBlockAppModule interface {
+	AppModule
```

Comment:
These should be on core/appmodule. We shouldn't introduce module related interface in types/module unless we need an sdk dependency